gsy_framework/utils.py

```diff
@@ -471,6 +471,13 @@ def convert_kW_to_kWh(power_W, slot_length):
     return convert_W_to_Wh(power_W, slot_length)
 
 
+# pylint: disable=invalid-name
+def convert_kWh_to_W(energy_kWh, slot_length):
+    """Convert W power into energy in Wh (based on the duration of the market slot)."""
+    energy_Wh = energy_kWh * 1000.0
+    return energy_Wh / (slot_length / duration(hours=1))
```

Comment:
which unit or type has the `slot_length` ? From the code, I see that it should be `pendulum.duration`, right?
If so, I propose 
```suggestion
def convert_kWh_to_W(energy_kWh: float, slot_length: duration) -> float:
    """Convert W power into energy in Wh (based on the duration of the market slot)."""
    energy_Wh = energy_kWh * 1000.0
    return energy_Wh / slot_length.total_hours()
```

Reply:
I added the type hints, however did not follow the `total_hours` suggestion, in order to be more evident on how the kWh -> W conversion is done here. Thanks!

Reply:
I just wanted to simplify the `(slot_length / duration(hours=1))`.
IMO we could use the features pendulum comes with. 
But if you do not like this, I am convinced to leave it.